userspace/libscap/scap_fds.c

```diff
@@ -491,6 +510,20 @@ uint32_t scap_fd_read_from_disk(scap_t *handle, OUT scap_fdinfo *fdi, OUT size_t
 		snprintf(handle->m_lasterr, SCAP_LASTERR_SIZE, "error reading the fd info from file, wrong fd type %u", (uint32_t)fdi->type);
 		return SCAP_FAILURE;
 	}
+
+	if(sub_len && *nbytes != sub_len)
+	{
+		toread = sub_len - *nbytes;
```

Comment:
I think an `ASSERT(*nbytes < sub_len)` (even potentially throwing an error) could be appropriate here, being new code. Same in the other places where you're using this snippet.

Reply:
I agree, will do